File: backend/src/Domain/Todo/Entities/Todo.php
<?php

namespace Domain\Todo\Entities;

class Todo
{
    private int $id;
    private string $title;
    private string $description;
    private string $dueDate;
    private string $priority;
    private ?string $image;
    private bool $isCompleted;
    
    public function __construct(
        string $title,
        string $description,
        string $dueDate,
        string $priority,
        ?string $image = null
    ) {
        $this->title = $title;
        $this->description = $description;
        $this->dueDate = $dueDate;
        $this->priority = $priority;
        $this->image = $image;
        $this->isCompleted = false;
    }
    
    // Add getters and setters
} 

File: mobile/lib/features/todo/domain/entities/todo.dart
class Todo {
  final int id;
  final String title;
  final String description;
  final DateTime dueDate;
  final String priority;
  final bool isCompleted;
  final String? image;
  final String? createdAt;
  final String? updatedAt;

  Todo({
    required this.id,
    required this.title,
    required this.description,
    required this.dueDate,
    required this.priority,
    required this.isCompleted,
    this.image,
    this.createdAt,
    this.updatedAt,
  });
}
